File: examples/go/greeter_test.go
package greeter

import "testing"

func TestHello(t *testing.T) {
	if got, want := Hello("Bazel"), "Hello, Bazel!"; got != want {
		t.Fatalf("Hello(\"Bazel\") = %q, want %q", got, want)
	}
}

func TestHelloDefault(t *testing.T) {
	if got, want := Hello(""), "Hello, world!"; got != want {
		t.Fatalf("Hello(\"\") = %q, want %q", got, want)
	}
}

File: examples/python/greeter_test.py
import unittest

from examples.python.greeter import hello


class GreeterTest(unittest.TestCase):
    def test_hello(self) -> None:
        self.assertEqual(hello("Bazel"), "Hello, Bazel!")

    def test_hello_default(self) -> None:
        self.assertEqual(hello(""), "Hello, world!")


if __name__ == "__main__":
    unittest.main()
